src/lib/statistics/cardinality_estimator.hpp

```diff
@@ -66,6 +68,26 @@ class CardinalityEstimator {
   std::shared_ptr<TableStatistics> estimate_statistics(const std::shared_ptr<const AbstractLQPNode>& lqp,
                                                        const bool cacheable, StatisticsByLQP& statistics_cache) const;
 
+  /**
+   * We use dummy objects for pruned statistics and cases where we do not estimate statistics (e.g., for aggregations).
```

Comment:
I thought about the same and simply used `Dummy` because we also use it in the `Projection` operator (`DummyTable` there). It's certainly not a mock. `EmptyStatistics` is also fine from my side. No strong opinion.